File: ALDS/1_D.js
'use strict';

function Main(input) {
	input = input.split('\n').map(Number);
	const n = input[0];
	let min = input[1];
	let ans = input[2] - input[1];
	for (let i = 1; i < n; i++) {
		ans = Math.max(ans, input[i + 1] - min);
		min = Math.min(min, input[i + 1]);
	}
	console.log(ans);
}

Main(require('fs').readFileSync('/dev/stdin', 'utf8'));

File: ALDS1/1_D.py
n = int(input())

nums = []

for _ in range(n):
    nums.append(int(input()))

minv = nums[0]
maxv = -200000000000

for i in range(1, n):
    maxv = max(maxv, nums[i] - minv)
    minv = min(minv, nums[i])

print(maxv)
